numba/tests/test_unicode.py

```diff
@@ -357,6 +362,14 @@ def test_slice3_error(self):
                                          cfunc(s, sl),
                                          "'%s'[%d:%d:%d]?" % (s, i, j, k))
 
+    def test_zfill(self):
+        pyfunc = zfill_usecase
+        cfunc = njit(pyfunc)
+
+        for s in UNICODE_EXAMPLES:
+            for width in range(0, 20):
+                self.assertEqual(pyfunc(s, width), cfunc(s, width))
+
```

Comment:
You should probably also test, if the implementation can deal with negative numbers and what happens in this case.

Reply:
Also, make sure that the implementation fails on floats, I believe the python implementation will do that too.